### ISSUE TITLE
Improve math support in Fira

### ISSUE THREAD

**Issue by fred-wang** [mozilla/Fira#79]
I'm just opening this for the record. I can help if people are interested. There are essentially two points:

1) Better Unicode coverage for math symbols. You can see a list in http://www.w3.org/TR/xml-entity-names/#blocks . In particular, https://en.wikipedia.org/wiki/Mathematical_Alphanumeric_Symbols is important.

2) Add an OpenType MATH table, and especially the mathvariant subtable which allows to stretch e.g. parenthesis or radical and draw e.g. summation symbol bigger in displaystyle mode.

This would important for the MathML rendering in Firefox OS in order to write math formulas that have style consistent with the default font (see https://bugzilla.mozilla.org/show_bug.cgi?id=775060). Currently one has to bundle his own math fonts in Firefox OS app in order to write math formulas...

Two remarks:
- Math support is often done in a separate font e.g. "STIX Math"
- Math font are often "serif". I don't know if there is a plan to have a "Fira Serif" version, but otherwise I guess the "sansserif" characters in the Mathematical Alphanumeric Symbols block are a bit irrelevant.

For more info, see the references at https://wiki.mozilla.org/MathML:Open_Type_MATH_Table#References (fontforge support, example of math fonts etc)

**Comment by Schweinepriester** [comment]
+1

**Comment by jfkthame** [comment]
I'm not sure extending Fira for full OpenType math support is a good use of resources at this point. Might be better to push for inclusion of one of the available math fonts[1] in FxOS, so that math content can be displayed in any context.

[1]  STIX? Latin Modern? Asana? E.g. see http://fred-wang.github.io/MathFonts/. We should look at the various options used in conjunction with Fira, to see what works together visually. I don't think having a distinct (serif) face that sets math formulae apart from the default (sans-serif) body text is a bad thing.

**Comment by fred-wang** [comment]
I agree with Jonathan. I merely opened this bug for the long term if we want a math companion for Fira, but I'm not even sure the font format used for Fira accepts a OpenType MATH table.

For the short term, the plan is https://bugzilla.mozilla.org/show_bug.cgi?id=775060 which we should be able to consider again soon once James fixes bug 1011020

**Comment by facundoq** [comment]
As a side note, this is also important for markup/programming languages that support unicode source code (latex/julia/etc).

**Comment by tweh** [comment]
+1

It would be really nice to have a fully working opentype sans-serif math fonts for documents set in all sans.
